multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/UnbindServiceFromApplicationStep.java

```diff
@@ -37,4 +39,41 @@ protected String getStepErrorMessage(ProcessContext context) {
                                                                                                   .getName());
     }
 
+    private ApplicationServicesUpdateCallback getApplicationServicesUpdateCallback(ProcessContext context) {
+        return new UnbindServiceFromApplicationCallback(context);
+    }
+
+    public static class UnbindServiceFromApplicationCallback implements ApplicationServicesUpdateCallback {
+
+        private final ProcessContext context;
+
+        public UnbindServiceFromApplicationCallback(ProcessContext context) {
+            this.context = context;
+        }
+
+        @Override
+        public void onError(CloudOperationException e, String applicationName, String serviceName) {
+            context.getStepLogger()
+                   .infoWithoutProgressMessage(Messages.CHECK_IF_SERVICE_BINDING_BETWEEN_SERVICE_0_AND_APP_1_ALREADY_DELETED, serviceName,
+                                               applicationName);
+            CloudControllerClient client = context.getControllerClient();
+
+            try {
+                if (ServiceBindingUtil.getServiceBinding(client, applicationName, serviceName) == null) {
```

Comment:
Why you need to check again for existing bindings when this is made inside of `CloudControllerRestClientImpl::doUnbindServiceInstance` ?

Reply:
As agreed, we will check only Http Status 404, because unbind will get app guid, service guid, then binding
If any of these does not exist, the CC will return 404 and we will handle it without further calls.